plugins/moodbar/__init__.py

```diff
@@ -543,20 +608,41 @@ def enable(exaile):
     else:
         _enable(None, exaile, None)
 
+
 def _enable(eventname, exaile, nothing):
-    global ExaileModbar
-    ExaileModbar.readMod('')
-    ExaileModbar.setupUi()
-    event.add_callback(ExaileModbar.play_start, 'playback_track_start', player.PLAYER)
-    event.add_callback(ExaileModbar.play_end, 'playback_player_end', player.PLAYER)
+    _enable_main_moodbar(exaile)
+
+    event.add_callback(_enable_preview_moodbar, 'preview_device_enabled')
+    event.add_callback(_disable_preview_moodbar, 'preview_device_disabling')
+
+    try:
+        import previewdevice
```

Comment:
The `enabled_plugins` dictionary (thanks for pointing that out) looks sufficient; how about:
``` python
def _get_preview_plugin_if_active(exaile):
    previewdevice = exaile.plugins.enabled_plugins.get('previewdevice', none)
    return getattr(previewdevice, 'preview_plugin', none)

def _enable(eventname, exaile, nothing):
    _enable_main_moodbar(exaile)

    event.add_callback(_enable_preview_moodbar, 'preview_device_enabled')
    event.add_callback(_disable_preview_moodbar, 'preview_device_disabling')

    preview_plugin = _get_preview_plugin_if_active(exaile)
    if getattr(preview_plugin, 'hooked', false):
        _enable_preview_moodbar('', preview_plugin, none)

```